### DISCUSSION TITLE
Problem reading images via

### DISCUSSION THREAD

**Discussion by OtavioPalamoni** [isl-org/Open3D#6376] [Q&A] [answered]
Hello, 

I'm new to Open3D so this might be a beginner's question, but when trying to create a RGBD image using a 2D RGB image and a Depth Map generated by MiDaS I keep getting the same error. To read the images I'm using the following:

image_rgb = o3d.io.read_image("cube_resized.png")
image_depth = o3d.io.read_image('depth_cube_resized.png')

THis leads to the following error:

[Open3D WARNING] Read PNG failed: unable to parse header.
Traceback (most recent call last):
  File "/home/****/****/****/test.py", line 34, in <module>
    image_rgbd = o3d.geometry.RGBDImage.create_from_color_and_depth(image_rgb, image_depth, convert_rgb_to_intensity=False)
RuntimeError: [Open3D Error] (static std::shared_ptr<open3d::geometry::RGBDImage> open3d::geometry::RGBDImage::CreateFromColorAndDepth(const open3d::geometry::Image&, const open3d::geometry::Image&, double, double, bool)) /root/Open3D/cpp/open3d/geometry/RGBDImageFactory.cpp:21: Unsupported image format.

I tried using a 640x425 image as example and it worked, then I resized my image using cv2.resize and it worked. However doing the same resize for another image I get the same error for both my RGB and Depth images. I've tried another solutions I found online like setting the full path to the images, and changing the uint but nothing worked. 

Is there any limitations (dimension, file size, ...) to what images can I read ?

**Comment by saurabheights** [comment]
Please share the images so we can test. Also, share how you installed open3d, python version, operating system info.

**Reply by OtavioPalamoni** [reply]
I am using Conda to set up my environment. I configured Python 3.9.18 via 'conda install python=3.9.' The Open3D library was installed following [the guide](http://www.open3d.org/docs/release/getting_started.html), To create the RGBD images, I followed the Redwood example in [this documentation](http://www.open3d.org/docs/latest/tutorial/Basic/rgbd_image.html).

Below, I'm attaching my RGB image (sample.png) and the depth map generated by MiDaS (depth.png). I was unable to send the raw depth values due to the extension (.pfm).

My OS is Ubuntu 22.04.

![depth](https://github.com/isl-org/Open3D/assets/127215270/b6525cba-5a5c-4935-a796-d18832bc6bcb)
![sample](https://github.com/isl-org/Open3D/assets/127215270/efa529de-3fb6-4b4a-aa90-a81e3f813728)

**Reply by saurabheights** [reply]
@OtavioPalamoni You can upload a zipped file in google drive. However, I am pretty sure issue is in rgb and depth image having different size. You probably resized one and not the other one. 

I will recommend you to move to development version. See here - http://www.open3d.org/docs/latest/getting_started.html#development-version-pip

It has many bug fixes such as https://github.com/isl-org/Open3D/pull/6305/files#diff-bdc3b39d18d88bf079861cb993f0e7a1418aaac4f8babf6b966d1daa65f2a5dcR60 where you would have had clearer error message.

**Reply by OtavioPalamoni** [reply]
Strange, I just check both images sizes using opencv via 'image.shape' and they are the same (4000x3000). I created [this google drive ](https://drive.google.com/drive/folders/1nhlzjPQcj8JmyIsupOQJf7BZOhwaNMFr?usp=drive_link)folder where you can find all files and a .zip file containing both images and the raw depth file for download.

**Reply by saurabheights** [reply, marked as answer]
@OtavioPalamoni Your image is named `.png` but it has `jpeg` format. Open3d doesnt have functionality to check image metadata, it uses extension to use the appropriate decoder which is why it failed to decode image.

Furthermore, since image read failed, it returned a `0x0` size color image which didnt match depth image size.

See below how I noticed and verified using `file` and `convert`:-

![image](https://github.com/isl-org/Open3D/assets/11634769/06ec8409-fb59-4401-b248-318fdd6efd18)
![image](https://github.com/isl-org/Open3D/assets/11634769/b1bbad63-5376-4bec-9fb4-be08a2f387e8)

**Reply by OtavioPalamoni** [reply]
That was the problem, after I converted them via Opencv a reconstructions was able to be generated. Thank you for the help.
